src/datadog/grok/filters/keyvalue.rs

```diff
@@ -25,98 +26,152 @@ use super::super::{
                 FunctionArgument::Arg(Value::Bytes(ref bytes)) => {
-                    let quotes = String::from_utf8_lossy(bytes);
-                    if quotes.len() == 2 {
-                        let mut chars = quotes.chars();
-                        vec![(
-                            chars.next().expect("open quote"),
-                            chars.next().expect("closing quote"),
-                        )]
-                    } else if quotes.is_empty() {
-                        // default
-                        vec![('"', '"'), ('\'', '\''), ('<', '>')]
-                    } else {
-                        return Err(GrokStaticError::InvalidFunctionArguments(f.name.clone()));
+                    let pair = String::from_utf8_lossy(bytes);
+                    match pair {
+                        pair if pair.len() == 2 => {
+                            let mut chars = pair.chars();
+                            Ok(vec![(
+                                chars.next().expect("open quote"),
+                                chars.next().expect("closing quote"),
+                            )])
+                        }
+                        pair if pair.is_empty() => Ok(Vec::from(DEFAULT_QUOTES)),
+                        _ => Err(GrokStaticError::InvalidFunctionArguments(f.name.clone())),
                     }
                 }
-                _ => return Err(GrokStaticError::InvalidFunctionArguments(f.name.clone())),
+                _ => Err(GrokStaticError::InvalidFunctionArguments(f.name.clone())),
             }
         } else {
-            // default quotes
-            vec![('"', '"'), ('\'', '\''), ('<', '>')]
-        };
+            Ok(Vec::from(DEFAULT_QUOTES))
+        }?;
 
         let field_delimiters = if args_len > 3 {
             match f.args.as_ref().unwrap()[3] {
                 FunctionArgument::Arg(Value::Bytes(ref bytes)) => {
-                    vec![String::from_utf8_lossy(bytes).to_string()]
+                    let delimiter_str = String::from_utf8_lossy(bytes);
+                    let mut delimiter_chars = delimiter_str.chars();
+                    match (delimiter_chars.next(), delimiter_chars.as_str()) {
+                        (Some(left), right) if !right.is_empty() => {
+                            (left.to_string(), right.to_string())
+                        }
+                        (Some(single), "") => (single.to_string(), single.to_string()),
+                        _ => return Err(GrokStaticError::InvalidFunctionArguments(f.name.clone())),
+                    }
                 }
                 _ => return Err(GrokStaticError::InvalidFunctionArguments(f.name.clone())),
             }
         } else {
-            // default field delimiters
-            vec![" ".into(), ",".into(), ";".into()]
+            (
+                DEFAULT_DELIMITERS.0.to_string(),
+                DEFAULT_DELIMITERS.1.to_string(),
+            )
```

Comment:
Final wondering, and this is in bike shed territory for sure: Maybe first assign the individual characters, and lift the repeated `.to_string()` calls out of the condition, or even leave them as `char`s when they are passed on.